### PR TITLE
Have absolute path for conf-root

### PR DESCRIPTION
**Pull request by nanonyme**
This enables calling configure from build-root to work https://github.com/apache/buildstream-plugins/issues/28

### CODE REVIEW

File: src/buildstream_plugins/elements/autotools.yaml [hunk 1]
```diff
@@ -17,6 +17,11 @@ variables:
   # Project-wide extra arguments to be passed to `configure`
   conf-global: ''
 
+  # This needs to be overridden for build-dir to work
```
Comment (gtristan) [inline src/buildstream_plugins/elements/autotools.yaml:20]: Two things here:
* The sentence *"This needs to be overridden for build-dir to work command-subdir is for monorepos to work"* doesn't read very well and appears to be developer facing technical detail. As this yaml is rendered in [user facing documentation](https://apache.github.io/buildstream-plugins/elements/autotools.html), comments here should be user facing, e.g.:
  * `conf-root`: The directory where the configure script will be generated and used
  * `command-subdir`: The subdirectory of `%{build-root}` where commands will be run
* I think this should be positioned near the `build-dir` as they seem functionally related

Comment (nanonyme) [reply src/buildstream_plugins/elements/autotools.yaml:20]: Thanks. I will fix once I have confidence that this is sufficient to make build-dir work. I have MR open at https://gitlab.com/freedesktop-sdk/freedesktop-sdk/-/merge_requests/9438

Comment (nanonyme) [reply src/buildstream_plugins/elements/autotools.yaml]: Looks like build-dir unsetting now works different than before ("." vs "" ) so I will need to spend some time fixing freedesktop-sdk. I will comment back here when I get a passing pipeline.

**Comment by nanonyme** [comment]
I am concerned whether this may leak element name (which is part of build-root) into binaries. I will iterate more on whether we can have correct reldir in this case.

**Comment by nanonyme** [comment]
@gtristan does the test image not have realpath?

**Comment by gtristan** [comment]
> @gtristan does the test image not have realpath?

I don’t know, but I’m doubtful of relying on expansion of `realpath` in default upstream autotools build instructions

**Comment by gtristan** [comment]
> > @gtristan does the test image not have realpath?
> 
> I don’t know, but I’m doubtful of relying on expansion of `realpath` in default upstream autotools build instructions

Also given we use `/bin/sh` by default I wouldn’t want to introduce bash specific `$(command …)` expansion

**Comment by nanonyme** [comment]
I am having some trouble figuring out how to handle the relative path generation with all these interdependent variables. freedesktop-sdk mostly had generation as a single flat string, hence why it was easy.

**Comment by nanonyme** [comment]
Especially existence of conf-root and conf-cmd seem to make configuring autotools harder, not easier.

**Comment by nanonyme** [comment]
Also conf-root as a concept is just *wrong*. You run configure and autoreconf from different directory. build-dir is a commonly required concept (eg glibc cannot be built without it). And autogen typically cannot be run from another directory. Many autogen scripts also forbid running them from any other directory than the directory where autogen is in. So conf-root cannot ever be changed from `.` without breaking said projects....

File: src/buildstream_plugins/elements/autotools.yaml (hunk 1 above)
Comment (nanonyme) [reply src/buildstream_plugins/elements/autotools.yaml]: Looks like the suggested way is to leak element name into Makefile as per https://github.com/apache/buildstream-plugins/blob/master/tests/elements/autotools/elements/amhelloconfroot.bst#L14 so I'll drop trying to get relative paths done.

**Comment by nanonyme** [comment]
Okay. I think this is a lost cause. I will revert the previous MR. I think the autotools.yml in this repo is broken by design and cannot be fixed without breaking API.

**Pull request closed without merging**